lib/tasks/one_off_tasks/fix_location.rake

```diff
@@ -0,0 +1,39 @@
+namespace :fix_location do
+  desc "fix locations currently in the database"
+
+  task fix_locations: :environment do
+    jobs_with_locations
+  end
+
+  def jobs_with_locations
+    stackoverflow_jobs = Job.where('url LIKE ?', '%stackoverflow%')
+    stackoverflow_jobs.each do |job|
+      location = Geocoder.search(job.location)
+      normalize_location(job) if location.empty?
+    end
+  end
+
+  def normalize_location(job)
+    if job.title =~ /(\w*, \w*)/
+      location = job.title[/\(([^\)]+)\)/]
+      response = Geocoder.search(location)
+      if !response.empty?
+        job.update_attributes(location: location)
+      else
+        check_for_location(job)
+      end
+    elsif job.title.include?('Singapore')
+      job.update_attributes(location: "Singapore")
+    elsif job.title.include?('Hong Kong')
+      job.update_attributes(location: "Hong Kong")
+    elsif job.title.include?('remote')
+      job.update_attribute(remote: true)
+      job.update_attribute(location: "remote")
+    end
+  end
+
+  def check_for_location(job)
+    # search other spots for the location
```

Comment:
Let me know if you have any ideas for beefing up this rake task. What I'm doing currently is checking the title for the pattern (word, other_word) then grabbing the contents of that parentheses and searching geocode with that location. If the response comes back empty then I know it's not a valid location and so I'm gonna send it to another method to check somewhere else. I was thinking about possibly checking a second parentheses for that pattern and checking it? Do you have any thoughts on that? @hhoopes @brennanholtzclaw @erinnachen @rrgayhart